irksome/stage.py

```diff
@@ -109,20 +124,47 @@ def getFormStage(F, butch, u0, t, dt, bcs=None, splitting=None,
     num_stages = butch.num_stages
     num_fields = len(V)
 
+    # default to no basis transformation, identity matrix
+    if vandermonde is None:
+        vandermonde = np.eye(num_stages + 1)
+
     # s-way product space for the stage variables
     Vbig = reduce(mul, (V for _ in range(num_stages)))
     VV = TestFunction(Vbig)
-    UU = Function(Vbig)
+    # UU = Function(Vbig)
+    ZZ = Function(Vbig)
 
     # set up the pieces we need to work with to do our substitutions
-    u0bits, vbits, VVbits, UUbits = getBits(num_stages, num_fields,
-                                            u0, UU, v, VV)
+    # u0bits, vbits, VVbits, UUbits = getBits(num_stages, num_fields,
+    #                                         u0, UU, v, VV)
+
+    u0bits, vbits, VVbits, ZZbits = getBits(num_stages, num_fields,
+                                            u0, ZZ, v, VV)
 
     MC = MeshConstant(V.mesh())
     vecconst = np.vectorize(lambda c: MC.Constant(c))
+
     C = vecconst(butch.c)
     A = vecconst(butch.A)
 
+    veccorz = np.vectorize(lambda c: ConstantOrZero(c, MC))
+    Vander = veccorz(vandermonde)
+    Vander_inv = veccorz(np.linalg.inv(vandermonde))
+
+    # multiply [u0bits; UUbits] = Vander @ [u0bits; UUbits]
```

Comment:
Because of the structure of the Bernstein Vandermonde matrix, the first entry is unchanged, and plus it's not part of the linear system we're solving for.

Reply:
I'm expanding the comments a bit more.

Reply:
And the commentary was wrong, anyway.